Some issues:

```
17:14:26 daemon/listeners/listeners_linux.go:65:43: too many arguments in call to activation.TLSListeners
17:14:26 	have (bool, *tls.Config)
17:14:26 	want (*tls.Config)
17:14:26 daemon/listeners/listeners_linux.go:67:40: too many arguments in call to activation.Listeners
17:14:26 	have (bool)
17:14:26 	want ()
```

--- a/daemon/listeners/listeners_linux.go
+++ b/daemon/listeners/listeners_linux.go
@@ -62,9 +62,9 @@ func listenFD(addr string, tlsConfig *tls.Config) ([]net.Listener, error) {
 	)
 	// socket activation
 	if tlsConfig != nil {
-		listeners, err = activation.TLSListeners(false, tlsConfig)
+		listeners, err = activation.TLSListeners(tlsConfig)
 	} else {
-		listeners, err = activation.Listeners(false)
+		listeners, err = activation.Listeners()
 	}
 	if err != nil {
 		return nil, err

